src/player-actions/violence/reducers.ts

```diff
@@ -24,17 +23,25 @@ export function inCombat(state: boolean = false, action: AppActions): boolean {
   }
 }
 
-export function enemyResources(
-  state = resourceReducerDefaultState,
+export function enemyCombatant(
+  state = enemyCombatantReducerDefaultState,
   action: AppActions
-): Resources {
+): Enemy {
   switch (action.type) {
     case LOAD_ENEMY:
       return {
-        energy: action.enemy.energy,
-        max_energy: action.enemy.max_energy,
-        health: action.enemy.health,
-        max_health: action.enemy.max_health
+        ...action.enemy
+      };
+    case ALTER_ENEMY_HEALTH:
+      const newHealth = state.health + action.health;
+      return {
+        ...state,
```

Comment:
Highly recommend checking out https://immerjs.github.io/immer/docs/introduction which massively makes reducers more readable and has all the benefits of immutability (+ strongly typed etc...)... I believe it's included in the `redux-toolkit` as well

Reply:
here's a good short course on it too https://egghead.io/courses/immutable-javascript-data-structures-with-immer